File: exercism.io/python/dnd-character/dnd_character.py
import random
import math

class Character:
    def __init__(self):
        self.strength = self.ability()
        self.dexterity = self.ability()
        self.constitution = self.ability()
        self.intelligence = self.ability()
        self.wisdom = self.ability()
        self.charisma = self.ability()
        self.hitpoints = 10 + modifier(self.constitution)
    
    def ability(self):
        r = [random.randint(1, 6) for i in range(0, 4)]
        r.sort()
        r.pop(0)
        return sum(r)
        
            
        


def modifier(num):
    return math.floor((num - 10) / 2)

File: exercism.io/javascript/dnd-character/dnd-character.js
//
// This is only a SKELETON file for the 'DnD Character' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

export const abilityModifier = (num) => {
  if(num > 18){ throw new Error('Ability scores can be at most 18') }
  if(num < 3){ throw new Error("Ability scores must be at least 3") }
  return Math.floor((num - 10) / 2)
};

export class Character {

  constructor(){
    this.str = Character.rollAbility()
    this.dex = Character.rollAbility()
    this.const = Character.rollAbility()
    this.intel = Character.rollAbility()
    this.wis = Character.rollAbility()
    this.cha = Character.rollAbility()
  }

  static rollAbility() {
    let result = []
    let sum = 0

    for(let i = 0; i < 4; i++){
      let n = Math.floor(Math.random() * 6) + 1
      result.push(n)
      sum += n  
    }

    sum -= result.sort((a, b) => a - b)[0]
    return sum
  }

  get strength() {
    return this.str
  }

  get dexterity() {
    return this.dex
  }

  get constitution() {
    return this.const
  }

  get intelligence() {
    return this.intel
  }

  get wisdom() {
    return this.wis
  }

  get charisma() {
    return this.cha
  }

  get hitpoints() {
    return 10 + abilityModifier(this.constitution)
  }
}
